src/react/databrowser/objects/__tests__/ObjectUpload.test.jsx

```diff
@@ -0,0 +1,66 @@
+import * as s3object from '../../../actions/s3object';
+import { BUCKET_NAME } from '../../../actions/__tests__/utils/testUtil';
+import ObjectUpload from '../ObjectUpload';
+import React from 'react';
+import { reduxMount } from '../../../utils/test';
+
+describe('ObjectUpload', () => {
+    it('should render ObjectUpload component', () => {
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: true,
+            },
+        });
+
+        expect(component.find(ObjectUpload).getDOMNode()).not.toBe(null);
+    });
+
+    it('should render an empty ObjectUpload component if showFolderCreate equals to false', () => {
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: false,
+            },
+        });
+
+        expect(component.find(ObjectUpload).getDOMNode()).toBe(null);
+    });
+
+    it('should call closeObjectUploadModal if cancel button is pressed', () => {
+        const closeObjectUploadModalMock = jest.spyOn(s3object, 'closeObjectUploadModal');
+
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: true,
+            },
+        });
+
+        expect(closeObjectUploadModalMock).toHaveBeenCalledTimes(0);
+        component.find('button#object-upload-cancel-button').simulate('click');
+        expect(closeObjectUploadModalMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not call uploadFiles if upload button is pressed when no files are present', () => {
```

Comment:
The upload button should be disabled.

Reply:
done

Reply:
Ha, sorry I wanted you to check that the button was disabled.

Reply:
done